neuraxle/steps/data.py

```diff
@@ -184,4 +187,126 @@ def __init__(self, wrapped, seed=None):
         Pipeline.__init__(self, [
             TrainOnlyWrapper(DataShuffler(seed=seed)),
             wrapped
-        ])
\ No newline at end of file
+        ])
+
+
+class ZipData(NonFittableMixin, NonTransformableMixin, BaseStep):
+    """
+    Zip two data sources together. Pass the name of the sub data containers to merge in the current data container.
+
+    Code example:
+
+    .. code-block:: python
+
+        # 3d shape: (batch_size, time_steps, n_features)
+        # 2d shape: (batch_size, batch_features)
+        # 1d shape: (batch_size)
+        data_container = DataContainer(
+            data_inputs=data_inputs_3d,
+            expected_outputs=expected_outputs_3d
+        ).add_sub_data_container('1d_data_source', data_container_1d) \
+            .add_sub_data_container('2d_data_source', data_container_2d)
+
+        p = Pipeline([
+            ZipData(data_sources=['1d_data_source', '2d_data_source'])
+        ])
```

Comment:
data_sources: what is the type? This is my biggest question for this PR. It will need to be documented.